src/components/TempTodo/TempTodo.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { ErrorMessage } from '../../types/ErrorMessage';
+import { deleteTodos } from '../../api/todos';
+
+type Props = {
+  todo: Todo;
+  isLoading: boolean;
+  handleDelete: (todoId: number) => void;
+  showError: (errorType: ErrorMessage) => void;
+  hideError: () => void;
+};
+
+export const TempTodo: React.FC<Props> = React.memo(({
+  todo,
+  isLoading,
+  handleDelete,
+  showError,
+  hideError,
+}) => {
+  const [isEdited] = useState(false);
+  const [isWaiting, setIsWaiting] = useState(false);
+
+  const { title, completed, id } = todo;
+
+  function handleDeleteTodo(): void {
+    hideError();
+    setIsWaiting(true);
+
+    deleteTodos(id)
+      .then(() => {
+        handleDelete(id);
+      })
+      .catch(() => {
+        showError(ErrorMessage.Delete);
+        setIsWaiting(false);
+      });
+  }
+
+  return (
+    <div
+      className={cn('todo', {
+        completed,
+      })}
+    >
+      <label className="todo__status-label">
+        <input
+          type="checkbox"
+          className="todo__status"
+          checked={completed}
+        />
+      </label>
+
+      {isEdited
+        ? (
+          <form>
+            <input
+              type="text"
+              className="todo__title-field"
+              placeholder="Empty todo will be deleted"
+              value={title}
+            />
+          </form>
+        )
+        : (
+          <>
+            <span className="todo__title">
+              {title}
+            </span>
+
+            <button
+              type="button"
+              className="todo__remove"
+              onClick={handleDeleteTodo}
+            >
+              {'\u00d7'}
+            </button>
+          </>
+        )}
+
+      <div
```

Comment:
you don't need show all content & implement all events here, bc you deleted this todo when you will be have todo
So, use critical params for this component